### ISSUE TITLE
Compile error: Mismatch between FindM4RI.cmake and xorfind.cpp

### ISSUE THREAD

**Issue by finnhaedicke** [msoos/cryptominisat#228]
I try to build [cryptominisat 4.2.0](http://msoos.org/largefiles/cryptominisat-4.2.0.tar.bz2) with [m4ri-20140914](http://m4ri.sagemath.org/downloads/m4ri/m4ri-20140914.tar.gz) on Linux (Fedora 20, CMake 2.8.12.2). I manually installed m4ri to a local directory (`/tmp/m4ri-20140914`). m4ri is otherwise not installed on the system.
 
I build cryptominisat and get a compile error:
```
$ mkdir build && cd build
$ export CMAKE_PREFIX_PATH="/tmp/m4ri-20140914"
$ cmake ../cryptominisat4-4.2.0
$ make
...
.../cryptominisat4/xorfinder.cpp:28:23: fatal error: m4ri/m4ri.h: No such file or directory
 #include <m4ri/m4ri.h>
```

I could Identify the reason to a mismatch of FindM4RI.cmake, include_directories and xorfinder.cpp.

In [FINDM4RI.cmake](https://github.com/msoos/cryptominisat/blob/master/cmake/FindM4RI.cmake#L36) `M4RI_INCLUDE_DIRS` is set to `/tmp/m4ri-20140914/include/m4ri` but  [xorfinder.cpp](https://github.com/msoos/cryptominisat/blob/master/cryptominisat4/xorfinder.cpp#L28) includes `m4ri/m4ri.h` which can not be found in any include directory. I assume this problem is not visible on systems that have m4ri installed system-wide as `/usr/include` contains `m4ri/m4ri.h`.

I think the solution is to change `xorfinder.cpp` to include `m4ri.h` as the pkg-config file for m4ri also uses the include/m4ri directory:
```
# from lib/pkgconfig/m4ri.pc
includedir=${prefix}/include
Cflags: -I${includedir}/m4ri -g -O2  -mmmx -msse -msse2 -msse3 
```

My workaround is to overwrite the cache variable  `M4RI_INCLUDE_DIRS`:
```
  $ cmake -DM4RI_INCLUDE_DIRS="/tmp/m4ri-20140914/include" ../cryptominisat4-4.2.0
```

**Comment by msoos** [comment]
Wow, thanks, nice catch! I'm thinking how best to change `FindM4RI.cmake` so that it works in both cases. Do you have any idea? I'm willing to change `xorfinder.cpp` of course.

**Comment by finnhaedicke** [comment]
Personally I favor the current include style (library/header). But FindM4RI currently behaves similar to pkg-config and changing xorfinder.cpp is definitly the easier fix (considering different platforms). So I would go with that.

FYI, wie found this problem while integrating cryptominisat and STP into the metaSMT dependencies repository.

**Comment by msoos** [comment]
Hey! I hope this fixes it :) Thanks again and cheers!

**Comment by finnhaedicke** [comment]
Thanks for fixing. Do you plan to create an updated release of cryptominisat soon?

**Comment by finnhaedicke** [comment]
I tested the change and it does not solve the hole problem. Now the internal includes of M4RI no longer work.

```
[  2%] Building CXX object cmsat4-src/CMakeFiles/libcryptominisat4.dir/xorfinder.cpp.o
In file included from /tmp/deps/build/cryptominisat4-4.2.0/cryptominisat4/xorfinder.cpp:28:0:
/tmp/deps/m4ri-20140914/include/m4ri/m4ri.h:56:22: fatal error: m4ri/mzp.h: No such file or directory
 #include <m4ri/mzp.h>
```

My argument to change xorfinder to pkg-config of M4RI  was invalid. M4RI had a bug in their pkg-config ([1][], [2][]) which is now fixed in their SCM.

Sorry for the messup. You should revert the fix and I will have a look into `FindM4RI.cmake` to match the fixed pkg-config.

[1]: https://bitbucket.org/malb/m4ri/issue/60/the-pc-pkg-config-file-content-is
[2]: https://bitbucket.org/malb/m4ri/commits/77ae998a57292368feaee5b9062e6ad45f128f5f

**Comment by msoos** [comment]
Hey! Thanks, I reverted. I don't understand fully what you need to do with `FindM4RI.cmake`, but I'll be happy to accommodate changes that don't break TravisCI and my install :) If you make a pull request and TravisCI says OK, we are good to go. Also, you can just tell me what is needed and I'll do it.

Thanks again for looking into this,

Mate
PS: Updated release should come after I tested it on AWS. That should be soon, maybe end of next weekend.
